mock/mock_test.go

```diff
@@ -1476,6 +1476,17 @@ func Test_MockReturnAndCalledConcurrent(t *testing.T) {
 	wg.Wait()
 }
 
+func Test_Mock_NumberOfCalls(t *testing.T) {
```

Comment:
Could you add a test which covers the issue you fixed. I would imagine it would fail when run with `go test -race` and _without_ the fix you've added in this PR?

Reply:
Sure, added the test to demonstrate data race. 

Commenting out the mutex yields a data race with `go test -v -race -run Test_Mock_NumberOfCallsIsThreadSafe ./...`